collector/remote_fx.go

```diff
@@ -0,0 +1,362 @@
+	FramesSkippedPerSecondInsufficientClientResources  *prometheus.Desc
+	FramesSkippedPerSecondInsufficientNetworkResources *prometheus.Desc
+	FramesSkippedPerSecondInsufficientServerResources  *prometheus.Desc
+	GraphicsCompressionratio                           *prometheus.Desc
+	InputFramesPerSecond                               *prometheus.Desc
+	OutputFramesPerSecond                              *prometheus.Desc
+	SourceFramesPerSecond                              *prometheus.Desc
+}
+
+// NewRemoteFx ...
+func NewRemoteFx() (Collector, error) {
+	const subsystem = "remote_fx"
+	return &RemoteFxCollector{
+		// net
+		BaseTCPRTT: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "net_base_tcp_rtt"),
+			"Base TCP round-trip time (RTT) detected in seconds",
+			[]string{"session_name"},
+			nil,
+		),
+		BaseUDPRTT: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "net_base_udp_rtt"),
+			"Base UDP round-trip time (RTT) detected in seconds.",
+			[]string{"session_name"},
+			nil,
+		),
+		CurrentTCPBandwidth: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "net_current_tcp_bandwidth"),
+			"TCP Bandwidth detected in thousands of bits per second (1000 bps).",
+			[]string{"session_name"},
+			nil,
+		),
+		CurrentTCPRTT: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "net_current_tcp_rtt"),
+			"Average TCP round-trip time (RTT) detected in seconds.",
+			[]string{"session_name"},
+			nil,
+		),
+		CurrentUDPBandwidth: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "net_current_udp_bandwidth"),
+			"UDP Bandwidth detected in thousands of bits per second (1000 bps).",
+			[]string{"session_name"},
+			nil,
+		),
+		CurrentUDPRTT: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "net_current_udp_rtt"),
+			"Average UDP round-trip time (RTT) detected in seconds.",
+			[]string{"session_name"},
+			nil,
+		),
+		TotalReceivedBytes: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "net_total_received_bytes"),
+			"(TotalReceivedBytes)",
+			[]string{"session_name"},
+			nil,
+		),
+		TotalSentBytes: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "net_total_sent_bytes"),
+			"(TotalSentBytes)",
+			[]string{"session_name"},
+			nil,
+		),
+		UDPPacketsReceivedPersec: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "net_udp_packets_received_total"),
+			"Rate in packets per second at which packets are received over UDP.",
+			[]string{"session_name"},
+			nil,
+		),
+		UDPPacketsSentPersec: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "net_udp_packets_sent_total"),
+			"Rate in packets per second at which packets are sent over UDP.",
+			[]string{"session_name"},
+			nil,
+		),
+
+		//gfx
+		AverageEncodingTime: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "gfx_average_encoding_time"),
```

Comment:
Should have the unit as a suffix
```suggestion
			prometheus.BuildFQName(Namespace, subsystem, "gfx_average_encoding_time_seconds"),
```
Are we sure the raw data is actually the average and not a counter of total seconds, by the way?

Reply:
yes, its a GaugeValue